src/ValueRepresentation.js

```diff
@@ -41,6 +49,68 @@ class ValueRepresentation {
         return this._isExplicit;
     }
 
+    addValueAccessors(value) {
+        return value;
+    }
+
+    /**
+     * Replaces a tag with a Proxy which assigns value accessors based on the vr field
+     * of the tag being given to it. If the tag object does not have a vr or vr.type
+     * property, the proxy will look for the prop name in the natural name map.
+     * @param {any} tag object to add accessors to
+     * @returns {any} either the same object if no accessor needed, or a Proxy
+     */
+    static addTagAccessors(tag) {
+        if (
+            !tag.__hasTagAccessors &&
+            ValueRepresentation.hasValueAccessors(tag.vr?.type || tag.vr)
+        ) {
+            // We replace the tag with a Proxy which intercepts assignments to obj[valueProp]
+            // and adds additional overrides/accessors to the value if need be. If valueProp
+            // is falsy, we check target.vr and add accessors via a ValueRepresentation lookup.
+            // Specifically, this helps address the incorrect (though common) use of the library:
+            //   dicomDict.dict.upsertTag('00101001', 'PN', 'Doe^John'); /* direct string assignment */
+            //   dicomDict.dict['00081070'].Value = 'Doe^John\Doe^Jane'; /* overwrite with multiplicity */
+            //   ...
+            //   jsonOutput = JSON.serialize(dicomDict);
+            // or:
+            //   naturalizedDataset.OperatorsName = 'Doe^John';
+            //   jsonOutput = JSON.serialize(naturalizedDataset);
+            // Whereas the correct usage of the dicom+json model would be:
+            //   dicomDict.dict.upsertTag('00101001', 'PN', [{Alphabetic:'Doe^John'}]);
+            //   naturalizedDataset.OperatorsName = [{Alphabetic:'Doe^John'},{Alphabetic:'Doe^Jane'}];
+            // TODO: refactor with addAccessors.js in mind
+            const handler = {
```

Comment:
Could you define the const handler statically?  Creating a new instance of it every time isn't needed, as the values inside it are static.